Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy

```diff
@@ -140,9 +182,9 @@ properties.buildReportOrder.each{ buildReportFile ->
 	deletions.each { deleteRecord ->
 		deletionCount += deleteRecord.getAttributeAsList("deletedBuildOutputs").size()
 		deleteRecord.getAttributeAsList("deletedBuildOutputs").each{ deletedFile ->
-			buildOutputsMap.put(deletedFile, [buildReport, deleteRecord])
+			def (dataset, member) = getDatasetName(deletedFile)
+			buildOutputsMap.put(new DeployableArtifact(member, "DELETE") , [dataset, buildReport, deleteRecord])
```

Comment:
Hi @M-DLB, I am running into an issue with this change to turn `member` + `deployType` into the key of the map: In my test scenario with multiple build reports, I have a scenario, that:
1. Adds a new member DELETEME
```
**  Deployable files detected in /var/jenkins/tmp/BuildReport-18.json
   JENKINS.ZDAT.RETIRE.LOAD(EBUD03), LOAD
   JENKINS.ZDAT.RETIRE.LOAD(EBUD02), LOAD
   JENKINS.ZDAT.RETIRE.LOAD(EBUD0RUN), LOAD
   JENKINS.ZDAT.RETIRE.LOAD(EBUD01), LOAD
   JENKINS.ZDAT.RETIRE.JCL(ALLOCDS), TRANSFER
   JENKINS.ZDAT.RETIRE.JCL(RUNZUNIT), TRANSFER
   JENKINS.ZDAT.RETIRE.JCL(EBUD0RUN), TRANSFER
   JENKINS.ZDAT.RETIRE.JCL(EBUD1RUN), TRANSFER
   JENKINS.ZDAT.RETIRE.JCL(DELETEME), TRANSFER
```
2. Deletes the member in a later stage in the process:
```
**  Deleted files detected in /var/jenkins/tmp/BuildReport-21.json
   JENKINS.ZDAT.RETIRE.SOURCE(DELETEME)
   JENKINS.ZDAT.RETIRE.JCL(DELETEME)
```

The new implementation carries now multiple entries: 
```
   Creating shiplist record for build output JENKINS.ZDAT.RETIRE.JCL(DELETEME) with recordType COPY_TO_PDS.
   Creating shiplist record for build output JENKINS.ZDAT.RETIRE.JCL(DELETEME) with recordType DELETE_RECORD.
   Defining shiplist delete container for JENKINS.ZDAT.RETIRE.JCL(DELETEME).
```
In this case UCD packaging fails with a duplicated member message:
```
** Create version by running UCD buztool
zOS toolkit config   : /var/ucd-agent/ (7.2.1.0,20211011-0758)
zOS toolkit binary   : /var/ucd-agent/ (7.2.1.0,20211011-0758)
zOS toolkit data set : RATCFG.UCD.V7R2M1 (7.2.1.0,20211011-0758)
Reading parameters:
....Command : createzosversion2
....Component : retirementCalculatorGithub
....Version : 169_20230424.093335.033
....Shiplist file : /var/jenkins/workspace/zunit-retirementCalculator/BUILD-169/build.20230424.113124.031/shiplist.xml
....Buztool Properties File : /var/jenkins/workspace/zunit-retirementCalculator//retirementCalculator/retirementCalculator/application-conf/retirementCalculator.ucd.properties
....Output File:/var/jenkins/workspace/zunit-retirementCalculator/BUILD-169/build.20230424.113124.031/buztool.output
Verifying version
....Repository location : /var/ucd-agent/var/repository/retirementCalculatorGithub/169_20230424.093335.033
Pre-processing shiplist:

*! Error executing buztool
Shiplist pre-processing error: duplicated member 'DELETEME' of JENKINS.ZDAT.RETIRE.JCL
```

I am attaching the full pipeline console output here - 
[consoleOutput-Duplicate-Member-Error.log](https://github.com/IBM/dbb/files/11319748/consoleOutput-Duplicate-Member-Error.log)

Reply:
That should be now fixed. Can you check again with your testcase?